### PR TITLE
SMK-1137/url preview save character encoding

### PR DESCRIPTION
**Pull request by Misirlou**

### CODE REVIEW

File: src/LinkPreview/Parser/YoutubeParser.php [hunk 1]
```diff
@@ -116,7 +116,8 @@ public function parseLink()
     {
         $this->readLink();
         $link = $this->getLink();
-        $htmlData = $this->parseHtml($link->getContent());
+        $content = $link->getCharset() == 'utf-8' ? utf8_decode($link->getContent()) : $link->getContent();
```
Comment (ruicampos) [inline src/LinkPreview/Parser/YoutubeParser.php:119]: comparison needs to be case-insensitive.

Example: Vimeo sends header with ``Content-Type:text/html; charset=UTF-8``

**Review by ruicampos — CHANGES_REQUESTED** [review]
Review comment

File: src/LinkPreview/Model/Link.php [hunk 2]
```diff
@@ -53,6 +57,24 @@ public function __construct($url = null)
...
+    {
+        return $this->charset;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setCharset($charset)
+    {
+        $this->charset = strtolower($charset);
```
Comment (Misirlou) [inline src/LinkPreview/Model/Link.php:70]: saving the enconding as lower case, kept the lower case comparison

**Review by ruicampos — APPROVED** [review]
LGTM

**Pull request merged**